src/Components/Component.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <thread>
+#include <atomic>
+#include <mutex>
+#include <queue>
+#include <iostream>
+using namespace std;
+template<class queueType>
+class Component {
+public:
+	void receive(queueType input) {
+		if (active) {
+			lock_guard<mutex> lock(sign);
+			pushData(input);
+			condition.notify_all();
+		}
+	}
+	void run() {
```

Comment:
Please add comment for the function:
https://google.github.io/styleguide/cppguide.html#Function_Comments